src/main/java/com/conveyal/r5/analyst/cluster/PathResultSummary.java

```diff
@@ -0,0 +1,177 @@
+                }
+                // Add to the set of durations
+                durations.add(iteration.totalTime);
+                IterationDetails iterationDetails = new IterationDetails(
+                        iteration.departureTime,
+                        iteration.waitTimes,
+                        iteration.totalTime,
+                        itineraryIndex
+                );
+                // Add to the list of departure times
+                this.iterations.add(iterationDetails);
+            }
+            List<TransitLeg> transitLegs = new ArrayList<>();
+            for (int routeIndex = 0; routeIndex < pathTemplate.routes.size(); routeIndex++) {
+                transitLegs.add(new TransitLeg(transitLayer, pathTemplate.stopSequence, routeIndex));
+            }
+            itineraries.add(new Itinerary(
+                    pathTemplate.stopSequence.access,
+                    pathTemplate.stopSequence.egress,
+                    transitLegs,
+                    allIterations.size(),
+                    durations.min(),
+                    durations.max(),
+                    itineraryIndex
+            ));
+            itineraryIndex += 1;
+        }
+        // Sort the iterations (by departure time and itinerary index)
+        Collections.sort(iterations);
+    }
+
+    /**
+     * An itinerary taken from a path result, including access and egress mode, transit legs, duration range (seconds),
+     * and how many iterations this itinerary was used.
+     */
+    static class Itinerary {
+        public StreetTimesAndModes.StreetTimeAndMode access;
+        public StreetTimesAndModes.StreetTimeAndMode egress;
+        public List<TransitLeg> transitLegs;
+        public int iterationsOptimal;
+        public int minSeconds;
+        public int maxSeconds;
+
+        public int index;
+
+        Itinerary(
+                StreetTimesAndModes.StreetTimeAndMode access,
+                StreetTimesAndModes.StreetTimeAndMode egress,
+                List<TransitLeg> transitLegs,
+                int iterationsOptimal,
+                int minSeconds,
+                int maxSeconds,
+                int index
+        ) {
+            this.access = access;
+            this.egress = egress;
+            this.transitLegs = transitLegs;
+            this.iterationsOptimal = iterationsOptimal;
+            this.maxSeconds = maxSeconds;
+            this.minSeconds = minSeconds;
+            this.index = index;
+        }
+    }
+
+    /**
+     * String representations of the boarding stop, alighting stop, and route, with ride time (in seconds).
+     */
+    static class TransitLeg {
+        public String routeId;
+        public String routeName;
+        public int rideTimeSeconds;
+        public String boardStopId;
+        public String boardStopName;
+
+        public String alightStopId;
+        public String alightStopName;
```

Comment:
👍 Nice to have these Id and Name values in separate fields instead of the prior concatenated strings